addons/core/addon/services/browser-object.js

```diff
@@ -0,0 +1,23 @@
+import Service from '@ember/service';
+
+/**
+ * A simple service that wraps around the global window and
+ * document variables that allows us to mock them in testing
+ */
+export default class BrowserObjectService extends Service {
+  /**
+   * The global window object
+   * @returns {Window}
+   */
+  get window() {
+    return window;
+  }
+
+  /**
+   * The global document object
+   * @returns {Document}
+   */
+  get document() {
```

Comment:
Should this be expressed in terms of `this.window`?

Reply:
Any particular reason? They reference the same object so I don't have a preference

Reply:
The goal is the fewest possible global references.  You could even express this using a computed `alias` to be idiomatic:  https://api.emberjs.com/ember/release/functions/@ember%2Fobject%2Fcomputed/alias

Reply:
Updated it to just use `this.window`